packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts

```diff
@@ -0,0 +1,146 @@
+import { parse } from 'flatted';
+import type { IPinData, IWorkflowExecutionDataProcess } from 'n8n-workflow';
+import assert from 'node:assert';
+import { Container, Service } from 'typedi';
+
+import { ActiveExecutions } from '@/active-executions';
+import type { ExecutionEntity } from '@/databases/entities/execution-entity';
+import type { User } from '@/databases/entities/user';
+import type { WorkflowEntity } from '@/databases/entities/workflow-entity';
+import { ExecutionRepository } from '@/databases/repositories/execution.repository';
+import { WorkflowRepository } from '@/databases/repositories/workflow.repository';
+import { NotFoundError } from '@/errors/response-errors/not-found.error';
+import { TestDefinitionService } from '@/evaluation/test-definition.service.ee';
+import type { IExecutionDb, IExecutionResponse } from '@/interfaces';
+import { WorkflowRunner } from '@/workflow-runner';
+
+/**
+ * This service orchestrates the running of test cases.
+ * It uses the test definitions to find
+ * past executions, creates pin data from them,
+ * and runs the workflow-under-test with the pin data.
+ * TODO: Evaluation workflows
+ * TODO: Node pinning
+ * TODO: Collect metrics
```

Comment:
Do we have an entity representing the single run of a test case? Or how are the results shown to the user?

Reply:
It will be added in the subsequent PRs, yes. There will be TestRun entity with the aggregated collected metrics